### PR TITLE
feat(mechanics): Allow adding tokens to DataNode

### PR DESCRIPTION
**Pull request by tibetiroka**
**Feature:** This PR adds a non-const function to DataNode which appends a token. This effectively means that useful data nodes can now be constructed in classes other than DataFile.

## Feature Details
I used the new feature in a couple places in DataFile, but I couldn't get around the `friend DataFile` thing entirely (which is arguably an anti-pattern).

## UI Screenshots
N/A

## Usage Examples
This can be used to greatly simplify saving complicated structures, such as the ones constructed in #9114. Currently, I have to construct the exact string I want to save (which is a nightmare in terms of code, and is really easy to mess up), but with this PR it would be greatly simplified (hopefully 20-30 lines of good code instead of a 100 line mess) with DataWriter's node writing function.

## Testing Done
Passed the tests. No extra testing done.

### Automated Tests Added
N/A

## Performance Impact
N/A

### CODE REVIEW

File: source/DataNode.h [hunk 1]
```diff
@@ -46,6 +46,8 @@ class DataNode {
 	// Get the token at the given index. No bounds checking is done internally.
 	// DataFile loading guarantees index 0 always exists.
 	const std::string &Token(int index) const;
+	// Add a new token to the end of this node.
+	void AddToken(std::string token);
```
Comment (RisingLeaf) [inline source/DataNode.h:50]:
```suggestion
	void AddToken(std::string token = "");
```
convenience?

File: source/DataFile.cpp [hunk 2]
```diff
@@ -193,9 +193,9 @@ void DataFile::LoadData(const string &data)
 			// range, but it appears that some libraries do not handle that case
 			// correctly. So:
 			if(tokenPos == endPos)
-				node.tokens.emplace_back();
+				node.AddToken("");
```
Comment (RisingLeaf) [inline source/DataFile.cpp:196]:
```suggestion
				node.AddToken();
```

**Review by RisingLeaf — APPROVED** [review]
looks good

**Review by Hurleveur — APPROVED** [review]
Looks good code wise indeed

**Comment by quyykk** [comment]
Idk seems like this would be better done in the other PR if it is needed there because it's not a big change anyways.

**Comment by tibetiroka** [comment]
Yeah I could add it to that one. Idk where this stands on the "one change per pr" thing, but yeah, this is definitely a minor change.

**Comment by tibetiroka** [comment]
Moved to #9114.

**Pull request closed without merging**